ntbs-service/Services/NotificationChangesService.cs

```diff
@@ -86,106 +85,87 @@ private IEnumerable<AuditLog> GetCanonicalLogs(List<AuditLog> group)
+                    return new List<AuditLog> {clinicalDetailsUpdate};
                 }
                 // .. or, if sites are the only edited items, we want to "fake" a clinical details log entry
-                else
-                {
-                    var fakeClinicalDetailsLog = group.First().Clone();
-                    fakeClinicalDetailsLog.EntityType = nameof(ClinicalDetails);
-                    fakeClinicalDetailsLog.EventType = "Update";
-                    yield return fakeClinicalDetailsLog;
-                }
-
-                yield break;
+                var fakeClinicalDetailsLog = group.First().Clone();
+                fakeClinicalDetailsLog.EntityType = nameof(ClinicalDetails);
+                fakeClinicalDetailsLog.EventType = "Update";
+                return new List<AuditLog> {fakeClinicalDetailsLog};
             }
 
             // Comorbidities page contains an Immunosuppression model, which creates its own audit entry
             // These are nicely separate concepts, so we can actually have them be detailed out on their own.
             if (group.Any(log => log.EntityType == nameof(ImmunosuppressionDetails)))
             {
-                foreach (var item in group)
-                {
-                    yield return item;
-                }
-
-                yield break;
+                return group;
             }
 
             // Social risks page contains sub-models, which create their own audit entries
             var socialRisksUpdate = group.Find(log => log.EntityType == nameof(SocialRiskFactors));
             if (socialRisksUpdate != null)
             {
-                yield return socialRisksUpdate;
-                yield break;
+                return new List<AuditLog> {socialRisksUpdate};
             }
 
             // Travel and visitor details get saved together, but it actually makes sense to show them independently
             if (group.Count == 2
                 && group.Any(log => log.EntityType == nameof(TravelDetails))
                 && group.Any(log => log.EntityType == nameof(VisitorDetails)))
             {
-                foreach (var item in group)
-                {
-                    yield return item;
-                }
-
-                yield break;
+                return group;
             }
 
             // Transferring creates a flurry of related events.
             // For each of these events we just want to have a single item in the history view 
-            var transferAlertLog = group.Find(log => log.EntityType == nameof(TransferAlert)
-                                                     || log.EntityType == nameof(TransferRejectedAlert));
-            if (transferAlertLog != null)
+
+            // Rejection creates:
+            //   - TransferAlert - Update
+            //   - TransferRejectedAlert - Insert
+            if (group.Any(log => log.EntityType == nameof(TransferRejectedAlert)))
```

Comment:
For the sake of consistency, I think we should stick with the pattern of calling `Find` then doing a null check, rather than calling `Any` then `Find`.